src/main/java/com/fontys/api/service/TournamentService.java

```
@@ -75,11 +79,43 @@ public String deleteTournament(Integer id) {
         return Optional.empty();
     }
 
-    private void validateTournament(Integer id) throws InvalidAttributeValueException {
+    @Transactional
+    public String addTeamToTournament(Integer tournamentId, Integer teamId) throws IllegalArgumentException
+    {
+        Optional<Tournament> tournament = tournamentRepository.findById(tournamentId);
+        Optional<Team> team = teamRepository.findById(teamId);
+        if (tournament.isEmpty())
+        {
+            return "The tournament does not exist";
+        }
+        else if (team.isEmpty())
+        {
+            return "The team does not exist";
+        }
+        else
+        {
+            Tournament tournament1 = tournament.get();
+            Team team1 = team.get();
+            if(tournament1.getTeams().contains(team1))
+            {
+                return "The team already joined the tournament!";
+            }
+            if(tournament1.getNumberOfTeams() == tournament1.getTeams().size())
+            {
+                return "The tournament is currently filled with teams!";
```

Maybe we can return an error message that is more clear.
Maybe: "The tournament no longer has empty slots for this team"